### PR TITLE
Report multipopulation

### PR DESCRIPTION
**Pull request by jorblancoa**
**Description**

Supports reports multipopulation
- Calculates population info of each gid and passes the info to libsonatareport
- Reads spike filename from report.conf

Fixes https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-785

**Use certain branches for the SimulationStack CI**

CI_BRANCHES:LIBSONATA_REPORT_BRANCH=multipop,NEURODAMUS_CORE_BRANCH=report_multipop,PY_NEURODAMUS_BRANCH=report_multipop

### CODE REVIEW

**Review by WeinaJi — APPROVED** [review]
Looks good to me.

File: coreneuron/io/reports/sonata_report_handler.cpp [hunk 1]
```diff
@@ -55,14 +71,14 @@ void SonataReportHandler::register_report(const NrnThread& nt,
         if (!vars.size())
             continue;
 
-        sonata_add_node(config.output_path.data(),
-                        config.population_name.data(),
-                        config.population_offset,
-                        gid);
+        auto pop_info = get_population_info(gid);
```
Comment (pramodk) [inline coreneuron/io/reports/sonata_report_handler.cpp:74]: `nitpick`:

```suggestion
        const auto& pop_info = get_population_info(gid);
```
```suggestion
        auto pop_info = get_population_info(gid);
```

File: coreneuron/io/output_spikes.cpp [hunk 2]
```diff
@@ -166,15 +165,15 @@ static void output_spike_populations(
                                      spikevec_gid.size());
         return;
     }
-    int n_populations = population_name_offset.size();
+    int n_populations = spikes_info.population_info.size();
     for (int idx = 0; idx < n_populations; idx++) {
-        auto cur_pop = population_name_offset[idx];
-        std::string population_name = cur_pop.first;
-        int population_offset = cur_pop.second;
+        auto curr_pop = spikes_info.population_info[idx];
```
Comment (pramodk) [inline coreneuron/io/output_spikes.cpp:170]: `just nitpick`:

```suggestion
        const auto& curr_pop = spikes_info.population_info[idx];
```

**Review by pramodk — APPROVED** [review]
LGTM 
(left just few `nitpicks`)

**Comment by pramodk** [comment]
As this is depend on different branches, I will convert this to Draft until @WeinaJi / @jorblancoa gives green light for this.

> CI_BRANCHES:LIBSONATA_REPORT_BRANCH=multipop,NEURODAMUS_CORE_BRANCH=report_multipop,PY_NEURODAMUS_BRANCH=report_multipop

**Pull request merged**